scripts/prepare_docker_buildinfo.sh

```diff
@@ -0,0 +1,57 @@
+#!/bin/bash
+
+IMAGENAME=$1
+DOCKERFILE=$2
+ARCH=$3
+DOCKERFILE_TARGE=$4
+DISTRO=$5
+
+[ -z "$BUILD_SLAVE" ] && BUILD_SLAVE=n
+[ -z "$DOCKERFILE_TARGE" ] && DOCKERFILE_TARGE=$DOCKERFILE
+DOCKERFILE_PATH=$(dirname "$DOCKERFILE_TARGE")
+BUILDINFO_PATH="${DOCKERFILE_PATH}/buildinfo"
+BUILDINFO_VERSION_PATH="${BUILDINFO_PATH}/versions"
+
+[ -d $BUILDINFO_PATH ] && rm -rf $BUILDINFO_PATH
+mkdir -p $BUILDINFO_VERSION_PATH
+
+# Get the debian distribution from the docker base image
+if [ -z "$DISTRO" ]; then
+    DOCKER_BASE_IMAGE=$(grep "^FROM" $DOCKERFILE | head -n 1 | awk '{print $2}')
+    DISTRO=$(docker run --rm --entrypoint "" $DOCKER_BASE_IMAGE cat /etc/os-release | grep VERSION_CODENAME | cut -d= -f2)
+    [ -z "$DISTRO" ] && DISTRO=jessie
+fi
+
+DOCKERFILE_PRE_SCRIPT='# Auto-Generated for buildinfo 
+COPY ["buildinfo", "/usr/local/share/buildinfo"]
+RUN dpkg -i /usr/local/share/buildinfo/sonic-build-hooks_1.0_all.deb
+RUN cp -rf /usr/local/share/buildinfo/trusted.gpg.d/* /etc/apt/trusted.gpg.d/
```

Comment:
>cp -rf /usr/local/share/buildinfo/trusted.gpg.d/* /etc/apt/trusted.gpg.d/ [](start = 4, length = 73)

Is it better to install gpg file by `curl | sudo apt-key add -`.
There is an example in sonic-slave-buster/Dockerfile.j2

Reply:
It is not good enough to copy several time when build each docker image. But it should has less impact. I do not use it for several reasons as below:
1. The curl is not a required package, as a web package it has lot of dependent packages.  We want to control all packages including the curl itself.
2.  We may have several gpg files, and any one can add more.

---
In reply to: [539815009](https://github.com/Azure/sonic-buildimage/pull/5718#discussion_r539815009) [](ancestors = 539815009)

Reply:
When you enable this feature in future, I think you need a place to download the gpg files sometime during the build process.
I agree we should not install many packages inside a docker image just for a one time `curl`.
How about download outside `docker build`, and `COPY` into the image?


---
In reply to: [540038466](https://github.com/Azure/sonic-buildimage/pull/5718#discussion_r540038466) [](ancestors = 540038466,539815009)